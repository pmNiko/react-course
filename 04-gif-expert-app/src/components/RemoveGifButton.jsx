import PropTypes from 'prop-types'

export const RemoveGifButton = ({removeCategory}) => {
  return (
    <div style={{display: 'flex', alignItems: 'center'}} >
      <h3>Este titulo no contiene Gifs!</h3>
      <button
        onClick={removeCategory} 
        style={{
          background: 'red', color: 'white', height: '22px', width: '23px', fontSize: 12, 
          borderColor: 'transparent', marginLeft: 10, boxShadow: '0px 3px 5px rgba(0, 0, 0, 0.5)',
          cursor: 'pointer' 
        }}>X</button>
    </div>
  )
}

RemoveGifButton.propTypes = {
  removeCategory: PropTypes.func.isRequired
}