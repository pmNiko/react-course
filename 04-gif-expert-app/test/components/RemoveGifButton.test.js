import { fireEvent, render, screen } from "@testing-library/react"
import { RemoveGifButton } from "../../src/components/RemoveGifButton.jsx"

describe('Pruebas en <RemoveGifButton/>', () => {
    const removeCategory = jest.fn();

    test('El componente debe hacer match con el snapshopt', () => { 
        const {container} = render( <RemoveGifButton removeCategory={removeCategory}/>);
        
        expect( container ).toMatchSnapshot();
    })
    
    test('Contiene un mensaje en el h3 informando la falta de gifs', () => { 
        render( <RemoveGifButton removeCategory={removeCategory}/>);
        
        expect( screen.getByRole( 'heading', {level: 3} ) ).toBeDefined();
        expect( screen.getByRole( 'heading', {level: 3} ).innerHTML ).toEqual('Este titulo no contiene Gifs!')
    })

    
    test('La funcion removeCategory debe ser llamada al clickear el boton', () => { 
        const removeCategory = jest.fn();
        render( <RemoveGifButton removeCategory={removeCategory}/>);

        fireEvent.click( screen.getByRole('button', { name: 'X' }));

        expect( removeCategory ).toHaveBeenCalled();
        expect( removeCategory ).toHaveBeenCalledTimes(1);
        
    })
})