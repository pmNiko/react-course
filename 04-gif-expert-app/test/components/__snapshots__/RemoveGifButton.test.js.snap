// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <RemoveGifButton/> El componente debe hacer match con el snapshopt 1`] = `
<div>
  <div
    style="display: flex; align-items: center;"
  >
    <h3>
      Este titulo no contiene Gifs!
    </h3>
    <button
      style="background: red; color: white; height: 22px; width: 23px; font-size: 12px; border-color: transparent; margin-left: 10px; box-shadow: 0px 3px 5px rgba(0, 0, 0, 0.5); cursor: pointer;"
    >
      X
    </button>
  </div>
</div>
`;
